pkg/kf/buildpacks/builder_creator.go

```diff
@@ -24,24 +24,30 @@ import (
 	"github.com/buildpack/pack"
 )
 
-// BuilderCreator creates a new buildback builder. It should be created via
+// BuilderCreator creates a new buildback builder.
+type BuilderCreator interface {
+	// Create creates and publishes a builder image.
+	Create(dir, containerRegistry string) (string, error)
+}
+
+// builderCreator creates a new buildback builder. It should be created via
 // NewCreateBuilder.
-type BuilderCreator struct {
+type builderCreator struct {
 	f BuilderFactoryCreate
 }
 
 // BuilderFactory creates and publishes new builde image.
 type BuilderFactoryCreate func(flags pack.CreateBuilderFlags) error
 
 // NewBuilderCreator creates a new BuilderCreator.
-func NewBuilderCreator(f BuilderFactoryCreate) *BuilderCreator {
-	return &BuilderCreator{
+func NewBuilderCreator(f BuilderFactoryCreate) BuilderCreator {
```

Comment:
IIRC returning an interface when you have a concrete type is an anti-pattern in go.

I think the preferred way is to do a check like `var _ BuilderCreatorInterface = (*BuilderCreator)(nil)` that'll fail at compile time if the interfaces don't match.

Reply:
Is wire OK with this? I did this because it was fussy if you didn't.

Reply:
Also, it looks like this follows suit with the rest of the codebase (e.g. pkg/kf/secrets/secrets.go).

Reply:
Yeah, I'm guilty of committing a bunch of code that breaks this rule. I'm not sure about wire, I'm okay with breaking rules if it means we get the utility of wire over hand-writing.

Reply:
Sounds good